src/vcpkg/portlint.cpp

```diff
@@ -0,0 +1,264 @@
+            {"GPL-3.0", "GPL-3.0-only"},
+            {"GPL-3.0+", "GPL-3.0-or-later"},
+            {"GPL-3.0-with-autoconf-exception", "GPL-3.0-only WITH Autoconf-exception-3.0"},
+            {"GPL-3.0-with-GCC-exception", "GPL-3.0-only WITH GCC-exception-3.1"},
+            {"LGPL-2.0", "LGPL-2.0-only"},
+            {"LGPL-2.0+", "LGPL-2.0-or-later"},
+            {"LGPL-2.1", "LGPL-2.1-only"},
+            {"LGPL-2.1+", "LGPL-2.1-or-later"},
+            {"LGPL-3.0", "LGPL-3.0-only"},
+            {"LGPL-3.0+", "LGPL-3.0-or-later"},
+            {"Nunit",
+             "DEPRECATED: This license is based on the MIT license, except with an \"acknowledgement\" clause. That "
+             "clause makes it functionally equivalent to MIT with advertising (Free, but GPLv2/v3 incompatible)"},
+            {"StandardML-NJ", "SMLNJ"},
+            {"wxWindows",
+             "DEPRECATED: Use license expression including main license, \"WITH\" operator, and identifier: "
+             "WxWindows-exception-3.1"}};
+        Status status = Status::Ok;
+        auto& license = scf.core_paragraph->license.value_or_exit(VCPKG_LINE_INFO);
+        for (const auto dep_license : deprecated_licenses)
+        {
+            const auto index = license.find(dep_license.first.c_str());
+            if (index == std::string::npos)
+            {
+                continue;
+            }
+            const auto end_index = index + dep_license.first.size();
+            if (end_index < license.size() && license[end_index] != ' ')
+            {
+                continue;
+            }
+            if (Strings::starts_with(dep_license.second, "DEPRECATED"))
+            {
+                msg::println_warning(msg::format(msgLintDeprecatedLicenseExpressionWithoutReplacement,
+                                                 msg::package_name = scf.core_paragraph->name,
+                                                 msg::actual = dep_license.first)
+                                         .append_raw(dep_license.second.substr(StringLiteral("DEPRECATED:").size())));
+
+                status |= Status::Problem;
+            }
+            else if (fix == Fix::NO)
+            {
+                msg::println_warning(msgLintDeprecatedLicenseExpressionWithReplacement,
+                                     msg::package_name = scf.core_paragraph->name,
+                                     msg::actual = dep_license.first,
+                                     msg::new_value = dep_license.second);
+                status |= Status::Problem;
+            }
+            else
+            {
+                license.replace(index, dep_license.first.size(), dep_license.second.c_str());
+                status |= Status::Fixed;
+            }
+        }
+
+        return status;
+    }
+
+    Status check_portfile_deprecated_functions(Filesystem& fs, SourceControlFileAndLocation& scf, Fix fix)
```

Comment:
I think you can create an override that operates with the contents of `portfile.cmake` and make the function that takes the `Filesystem` and `SCFL` call the new function,

```cpp
struct FixedPortfile {
    Status status,
    Optional<std::string> contents, // empty if Fix::NO
    Optional<std::set<StringLiteral>> added_host_deps // host-dependencies that need to be added, empty if Fix::NO
};
FixedPortfile check_portfile_deprecated_functions(StringView contents, StringVIew origin, Fix fix, MessageSink& warningsSink);

Status check_portfile_deprecated_functions(Filesystem& fs, SourceControlFileAndLocation& scf, Fix fix)
{
    auto contents = fs.read_contents(scf.source_location / "portfile.cmake", VCPKG_LINE_INFO);
    auto lint_result = check_portfile_deprecated_functions(contents, scf.source_control_file->core_paragraph->name, fix, stdout_sink);
    // add host dependencies and write fixed portfile (or do nothing if Fix::NO)
    return lint_result.status;
}
```

Reply:
(also for the sake of unit tests)